### ISSUE TITLE
Basic / Paper UI: locale are not taken into account when formatting e.g. decimal values

### ISSUE THREAD

**Issue by MHerbst** [eclipse/smarthome#2714]
Decimal values are always formatted with the default locale. A different default locale set in the configuration file or in Paper UI under Configuration/System is ignored.

One example is method `format` of class `org.eclipse.smarthome.core.library.types.DecimalType`: this method calls several times `String.format` but in all calls the _locale_ parameter is missing and thus the default locale of the server system is used. This has an influence on the formatting of decimal values ("." vs. ",").

Same problem exists probably on other classes of this package too.

**Comment by watou** [comment]
> One example is method format of class org.eclipse.smarthome.core.library.types.DecimalType: this method calls several times String.format 

The string format of Commands and States should not be variable by locale.  Locale should matter at the UI layer, though.

**Comment by MHerbst** [comment]
@watou I agree with you. 

I thought that the "format" methods of the different type classes are only called in the UI context to format the strings according to label definitions of items resp. in sitemaps.

**Comment by rtvb** [comment]
The formatted uses the Java locale. Unless specified that is the OS locale. To overwrite this, 2 command line parameters can be set.

http://www.oracle.com/technetwork/articles/javase/locale-140624.html?ssSourceSiteId=ocomen#using

https://community.openhab.org/t/locale-specific-datetime-formatting/22656/4?u=rtvb

**Comment by MHerbst** [comment]
It is possible to solve it this way, but in my opinion this is only a bypass. It is possible to define the locale in Paper UI but this locale settings is ignored when formatting decimals and dates. And this is the problem.

**Comment by rtvb** [comment]
@MHerbst You are absolutely right. openHAB should respect the locale setting. But it would be impossible to develop a custom made locale handler for low level functions. That implies that setting the locale in openHAB should set the Java locale.

A possible solution would be, to require a restart after the user has changed country and/or language. openHAB should then automatically add both parameters `user.country` and `user.language` to the startup command line, if specified in the configuration.

**Comment by maggu2810** [comment]
Eclipse SmartHome has its own locale provider (available as an OSGi service).
So you can set the and get the locale that should be used by the Eclipse SmartHome ecosystem without changing the user locale settings and there is also no need to change the locale of the whole JVM.

**Comment by maggu2810** [comment]
* https://github.com/eclipse/smarthome/blob/bf5539767b8e92aee9ebfd3d2a703952a1a43ede/bundles/core/org.eclipse.smarthome.core/OSGI-INF/localeprovider.xml
* https://github.com/eclipse/smarthome/blob/bf5539767b8e92aee9ebfd3d2a703952a1a43ede/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/i18n/LocaleProvider.java
* https://github.com/eclipse/smarthome/blob/bf5539767b8e92aee9ebfd3d2a703952a1a43ede/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/i18n/LocaleProviderImpl.java

In the PaperUI it could be configured using "Configuration, System, Locale".

If could also be set by the standard ESH service configuration files.
bf5539767b8e92aee9ebfd3d2a703952a1a43ede

**Comment by rtvb** [comment]
@maggu2810 then why is my nl/NL setting not working? I see English weekday and month names and dots for decimals instead of commas.

**Comment by maggu2810** [comment]
I don't say that the setting is used in every case and I don't know if it is bug free.
I just want to give you some background information where you can look at to find what is going wrong.

**Comment by MHerbst** [comment]
@rtvb the cause of the problem is probably here https://github.com/eclipse/smarthome/blob/f7aa4bc9787f6e90f94f924b7efc10ca382e8e86/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/DecimalType.java#L74
and here https://github.com/eclipse/smarthome/blob/f7aa4bc9787f6e90f94f924b7efc10ca382e8e86/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/DateTimeType.java#L75

The format method is called without specifying a locale. So the always the default locale is used.

I did not have the time to figure out the best way to get the configured locale in these classes otherwise I already tried to fix it.
